.circleci/config.yml

```diff
@@ -99,6 +100,145 @@ jobs:
+          paths:
+            - ./node_modules
+      - save_cache:
+          key: android-{{ checksum ".circleci/config.yml" }}-{{ checksum "android/build.gradle" }}-{{ checksum  "android/app/build.gradle" }}
+          paths:
+            - ~/.gradle
+      - run:
+          command: |
+            yes | sdkmanager --licenses
+            yes | sdkmanager "platform-tools" "tools" >/dev/null
+            yes | sdkmanager "platforms;android-27" "system-images;android-27;google_apis;x86" >/dev/null
+            yes | sdkmanager "emulator" --channel=3 >/dev/null
+            yes | sdkmanager "build-tools;27.0.3" >/dev/null
+            yes | sdkmanager --licenses >/dev/null
+            yes | sdkmanager --list
+          name: Install Android Emulator
+          shell: /bin/bash -e
+      - run:
+          name: Detox Build Android App
+          command: |
+            source ~/.bash_profile
+            detox build --configuration android.emu.release
+      - run:
+          command: |
+            source ~/.bash_profile
+            adb start-server
+            adb devices
+            adb kill-server
+          name: ADB Start Stop
+      - run:
+          command: |
+            source ~/.bash_profile
+            avdmanager create avd --force -n Pixel_3_XL_API_27 -k "system-images;android-27;google_apis;x86" -g google_apis -d "Nexus 4"
+          name: Create Android Emulator
+      - run:
+          background: true
+          command: |
+            /usr/local/share/android-sdk/emulator/emulator @Pixel_3_XL_API_27 -version
+            /usr/local/share/android-sdk/emulator/emulator @Pixel_3_XL_API_27 -skin 1440x2880 -cores 2 -gpu auto -accel on -memory 4096 -no-audio -no-snapshot -no-boot-anim -no-window -logcat *:W | grep -i 'ReactNative\|rocket'
+          name: Start Android Emulator (background)
+      - run:
+          command: |
+            source ~/.bash_profile
+            export BOOT=""
+            echo "Waiting for AVD to finish booting"
+            export PATH=$(dirname $(dirname $(command -v android)))/platform-tools:$PATH
+            until [[ "$BOOT" =~ "1" ]]; do
+              sleep 5
+              export BOOT=$(adb -e shell getprop sys.boot_completed 2>&1)
+            done
+            sleep 15
+            adb shell settings put global window_animation_scale 0
+            adb shell settings put global transition_animation_scale 0
+            adb shell settings put global animator_duration_scale 0
+            echo "Android Virtual Device is now ready."
+          name: Wait for AVD to be ready
+          no_output_timeout: 5m
+      - run:
+          command: |
+            source ~/.bash_profile
+            /usr/local/share/android-sdk/platform-tools/adb -s emulator-5554 install -r -g -t "/Users/distiller/project/android/app/build/outputs/apk/release/app-release.apk"
+            detox test -c android.emu.release -l trace --headless
```

Comment:
Why `trace --headless`?

Reply:
removed loglevel trace. Headless is suggested by detox cli `[Android Only] Launch Emulator in headless mode. Useful when running on CI.` and it should be a bit faster this way